cmd/kpod/ps.go

```diff
@@ -0,0 +1,354 @@
+	if err != nil {
+		return errors.Wrapf(err, "could not get list of containers matching filter")
+	}
+
+	return psOutput(containerList, server, opts)
+}
+
+func psOutput(containers []*libkpod.ContainerData, server *libkpod.ContainerServer, opts psOptions) error {
+	var (
+		output           []psOutputParams
+		containersOutput []*libkpod.ContainerData
+		status           string
+		ctrID            string
+		command          string
+		runningFor       string
+		imageName        string
+		mounts           string
+		ports            string
+		size             string
+	)
+
+	if len(containers) > 0 {
+		if opts.latest {
+			containersOutput = append(containersOutput, containers[0])
+		} else if opts.last > 0 {
+			if opts.last > len(containers) {
+				containersOutput = containers
+			} else {
+				for i := 0; i < opts.last; i++ {
+					containersOutput = append(containersOutput, containers[i])
+				}
+			}
+		} else if !opts.all {
+			for _, ctr := range containers {
+				if ctr.State.Status == "running" {
+					containersOutput = append(containersOutput, ctr)
+				}
+			}
+		} else {
+			containersOutput = containers
+		}
+
+		for _, ctr := range containersOutput {
+			ctrID = ctr.ID
+			runningFor = units.HumanDuration(time.Since(ctr.State.Created))
+			command = getCommand(ctr.ImageCreatedBy)
+			imageName = getImageName(ctr.FromImage, opts.noTrunc)
+			mounts = getMounts(ctr.Mounts, opts.noTrunc)
+			ports = getPorts(ctr.Config.ExposedPorts)
+			size = units.HumanSize(float64(ctr.SizeRootFs))
+
+			if ctr.State.Status == "stopped" {
+				status = "Exited (" + strconv.FormatInt(int64(ctr.State.ExitCode), 10) + ") " + runningFor + " ago"
+			} else if ctr.State.Status == "running" {
+				status = "Up " + runningFor + " ago"
+			} else {
+				status = "Created"
+			}
+
+			if !opts.noTrunc {
+				ctrID = ctr.ID[:idTruncLength]
+			}
+
+			params := psOutputParams{
+				ID:         ctrID,
+				Image:      imageName,
+				Command:    command,
+				CreatedAt:  ctr.State.Created,
+				RunningFor: time.Since(ctr.State.Created),
+				Status:     status,
+				Ports:      ports,
+				Size:       size,
+				Names:      ctr.Name,
+				Labels:     ctr.Labels,
+				Label:      "",
+				Mounts:     mounts,
+			}
+			output = append(output, params)
+		}
+	}
+	var out formats.Writer
+
+	switch opts.format {
+	case "":
+		out = outStruct{output: output, opts: opts}
+	case "json":
+		out = formats.JSONstruct{Output: toGenericPS(output)}
+	default:
+		out = formats.StdoutTemplate{Output: toGenericPS(output), Template: opts.format}
+	}
+
+	formats.Writer(out).Out()
+	return nil
+}
+
+// getCommand gets the actual command from the whole command
+func getCommand(cmd string) string {
+	start := strings.Index(cmd, "[")
+	if start == -1 {
+		return ""
+	}
+	start += len("[")
+	end := strings.Index(cmd, "]")
+	return cmd[start:end]
+}
+
+// getImageName shortens the image name when noTrunc is false
+func getImageName(img string, noTrunc bool) string {
```

Comment:
The image name is shortened as well, for example if it is from Dcoker hub you want to strip off the "docker.io/library".